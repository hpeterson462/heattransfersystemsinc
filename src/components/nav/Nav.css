.navbar {
  display: flex;
  flex-wrap: wrap;
  position: fixed;
  overflow: hidden;
  width: 100%;
  top: 0;
  left: 0;
  justify-content: space-evenly;
  align-items: center;
  overflow: hidden;
  z-index: 100;
  box-shadow: 0px 3px 10px rgba(0, 0, 0, 0.1);
  background-color: white;
}

.logo-icon {
  width: 4rem;
}

.logo-icon:hover {
  cursor: pointer;
}

.link {
  font-size: 1rem;
  text-transform: uppercase;
  text-decoration: none;
  padding: 1rem;
  color: #003366;
  transition: 0.45s ease-in-out;
}

.link::after {
  position: absolute;
  content: "";
  width: 100%;
  height: 3px;
  top: 100%;
  left: 0;
  background: #F68D0B;
  transition: transform 0.5s;
  transform: scaleX(0);
}

.link:hover {
  cursor: pointer;
  color: #F68D0B;
  transform: scale(1.0);
}

.link:hover::after {
  transform: scaleX(0.7);
}

@media only screen
and (min-device-width : 20em) 
and (max-device-width : 50em)
and (-webkit-min-device-pixel-ratio: 3)
and (orientation: portrait) {
  .navbar {
    transition: top 0.3s;
  }
  .logo-icon {
    display: none;
  }
}
