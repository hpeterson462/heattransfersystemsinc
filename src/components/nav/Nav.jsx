import React, { useState, useEffect } from 'react';
import { HashLink as Link } from 'react-router-hash-link';
import logo from '../../assets/logo.png';
import './Nav.css';

export default function Nav() {

  // const [startScrollPosition, setStartScrollPosition] = useState(0);
  // const [visible, setVisible] = useState(true);

  // const handleScroll = () => {
  //   const currentScrollPosition = window.pageYOffset;

  //   setVisible((startScrollPosition > currentScrollPosition && startScrollPosition - currentScrollPosition) || currentScrollPosition < 10);

  //   setStartScrollPosition(currentScrollPosition);
  // };

  // useEffect(() => {
  //   window.addEventListener('scroll', handleScroll);

  //   return () => window.removeEventListener('scroll', handleScroll);

  // }, [startScrollPosition, visible, handleScroll]);

  return (
    // <nav className="navbar" style={{ top: visible ? '0' : '-150px' }}>
    <nav className="navbar">

      <Link to="#home">
        <img
          className="logo-icon"
          src={logo}
          alt="small logo icon"
        />
      </Link>

      <Link className="link" to="#home">
        Home
      </Link>

      <Link className="link" to="/news">
        News
      </Link>

      <Link className="link" to="#solutions">
        Solutions
      </Link>

      <Link className="link" to="#services">
        Services
      </Link>

      <Link className="link" to="#products">
        Products
      </Link>

      <Link className="link" to="#aboutus">
        About Us
      </Link>

      <Link className="link" to="#contact">
        Contact
      </Link>
    </nav>
  )
}
