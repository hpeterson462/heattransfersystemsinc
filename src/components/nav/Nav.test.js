import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Nav from './Nav';
import { getByText, toBeInTheDocument } from '@testing-library/jest-dom';

describe('Nav component', () => {
  afterEach(() => cleanup());

  it('renders Nav component', () => {
    const { asFragment } = render(<MemoryRouter><Nav /></MemoryRouter>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders nav link', () => {
    render(<MemoryRouter><Nav /></MemoryRouter>);

    expect(screen.getByText(/Solutions/)).toBeInTheDocument();
  });
});
