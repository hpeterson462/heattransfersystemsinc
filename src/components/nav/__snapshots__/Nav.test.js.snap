// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders Nav component 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
    style="top: 0px;"
  >
    <a
      href="/#home"
    >
      <img
        alt="small logo icon"
        class="logo-icon"
        src="logo.png"
      />
    </a>
    <a
      class="link"
      href="/#home"
    >
      Home
    </a>
    <a
      class="link"
      href="/#solutions"
    >
      Solutions
    </a>
    <a
      class="link"
      href="/#services"
    >
      Services
    </a>
    <a
      class="link"
      href="/#products"
    >
      Products
    </a>
    <a
      class="link"
      href="/#aboutus"
    >
      About Us
    </a>
    <a
      class="link"
      href="/#contact"
    >
      Contact
    </a>
  </nav>
</DocumentFragment>
`;
